src/Makefile.propagate

```diff
@@ -15,6 +15,7 @@ test: $(NEAL) $(patsubst %/Makefile,%.test,$(SUBDIRS))
 	@true
 
 %.build:
+	[ "$*" != "providers" ] || sleep 1
```

Comment:
this is sleeping on `make providers`?
can you explain a bit more on what's the race condition?

Reply:
The `src/core` module gets compiled and is then installed as a local package `neal` for opam ( OCaml Package Manager ) inside the `register` task here: https://github.com/uber/NEAL/blob/master/src/Makefile. 

However, the `src/providers` which depend on this local `neal` package starts to build as soon as the build task for `neal` is complete and  thus complains about `Could not find module: neal`

Reply:
sounds more like a dependency chain setup issue
ok with this for now, would rather fix the makefile if possible
(is 1s good enough?)

Reply:
Ya, "RECURSIVE MAKE CONSIDERED HARMFUL" https://accu.org/journals/overload/14/71/miller_2004/